torch_xla/core/dynamo_bridge.py

```diff
@@ -111,6 +111,65 @@ def get_fallback_ops():
   return fallback_ops
 
 
+# Checks that all input args that are tensors are on the same device.
+def _get_input_arg_device(input_args: tuple) -> torch.device:
+  device = None
+  for arg in input_args:
+    if not isinstance(arg, torch.Tensor):
+      continue
+
+    if device is None:
+      device = arg.device
+    else:
+      assert arg.device == device, "Not all args are on the same device."
+
+    return device
+
+
+# Returns True if all the input args are on a CUDA device.
+def _args_on_cuda(input_args: tuple) -> bool:
+  input_device: torch.device = _get_input_arg_device(input_args)
+  if input_device is None:
+    return False
+
+  return input_device.type == "cuda"
+
+
+# Given an input list, moves the tensors to the given target_device.
+# The output order will be the same as the input. Non tensors will also still
+# be in the list.
+def _maybe_move_tensors_to_device(tensors: tuple,
+                                  target_device: torch.device) -> tuple:
+  if not torch.cuda.is_available():
+    return tensors
```

Comment:
Why do we need this?

Reply:
This works with CPU, too, doesn't it?

Reply:
We have had a couple discussion about whether to support CPU. My concern is that moving tensors between devices will force the execution to be blocking. The main motivation for this pr was to support execute the part that dynamo can't capture on cuda tensor, so I want to limit the scope of this pr to CUDA tensor only for now. We can support CPU in the future as well.

Reply:
Correct and I'm just being defensive here to ensure someone doesn't accidentally call this and it supports non-CUDA moves.